electron-app/src/server/websites/e621/e621.service.ts

```diff
@@ -165,15 +180,96 @@ export class e621 extends Website {
+
   async validateFileSubmission(
     submission: FileSubmission,
     submissionPart: SubmissionPart<any>,
     defaultPart: SubmissionPart<DefaultOptions>,
   ): Promise<ValidationParts> {
+    const oldToken = this.validateSubmissionCancelTokens.get(submission._id);
+    if (oldToken) oldToken.cancel();
+
+    const cancellationToken = new CancellationTokenDynamic();
+    this.validateSubmissionCancelTokens.set(submission._id, cancellationToken);
+
     const problems: string[] = [];
     const warnings: string[] = [];
     const isAutoscaling: boolean = submissionPart.data.autoScale;
 
+    try {
+      await this.validateTags(defaultPart, submissionPart, problems, warnings, cancellationToken);
+      await this.validateUserFeedback(submissionPart, warnings, cancellationToken);
+    } catch (e) {
+      // Ignore validation canceled errors
+      if (!(e instanceof CancellationException)) throw e;
+    }
+
+    if (!WebsiteValidator.supportsFileType(submission.primary, this.acceptsFiles)) {
+      problems.push(`Currently supported file formats: ${this.acceptsFiles.join(', ')}`);
+    }
+
+    const { type, size, name } = submission.primary;
+
+    if (FileSize.MBtoBytes(this.MAX_MB) < size) {
+      if (
+        isAutoscaling &&
+        type === FileSubmissionType.IMAGE &&
+        ImageManipulator.isMimeType(submission.primary.mimetype)
+      ) {
+        warnings.push(`${name} will be scaled down to ${this.MAX_MB}MB`);
+      } else {
+        problems.push(`e621 limits ${submission.primary.mimetype} to ${this.MAX_MB}MB`);
+      }
+    }
+
+    return { problems, warnings };
+  }
+
+  private async validateUserFeedback(
+    submissionPart: SubmissionPart<any>,
+    warnings: string[],
+    cancellationToken: CancellationTokenDynamic,
+  ) {
+    try {
+      const username = this.getAccountInfo(submissionPart.accountId, 'username');
+      const feedbacks = await this.getUserFeedback(cancellationToken, username);
+
+      if (Array.isArray(feedbacks)) {
+        for (const feedback of feedbacks) {
+          if (feedback.category === e621UserFeedbackCategory.Positive) continue;
+
+          const updatedAt = new Date(feedback.updated_at);
+          const week =
+            /*ms*/ 1000 * /*sec*/ 60 * /*min*/ 60 * /*hour*/ 60 * /*day*/ 24 * /*week*/ 7;
+
+          if (Date.now() - updatedAt.getTime() > week) continue;
+
+          warnings.push(
+            `You have recent ${
+              feedback.category
+            } feedback at the https://e621.net/user_feedbacks?search[user_name]=${username}. Feedback: ${
+              feedback.body.length > 100
+                ? `${feedback.body.slice(0, 100)}... (More on the e621)`
+                : feedback.body
+            }`,
+          );
+        }
+      }
+    } catch (error) {
+      this.logger.error(error);
+      warnings.push(`Unable to get user feedback. You can check your account manually`);
+    }
+  }
+
```

Comment:
validateTags function and its subfunctions are so big but i was not sure if i should move them to the separate class even tho they have their own "context"